wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj2/command/CommandScheduleTest.java

```diff
@@ -109,6 +111,102 @@ void schedulerCancelTest() {
+  void cancelNextCommandTest() {
+    try (CommandScheduler scheduler = new CommandScheduler()) {
+      Command[] commands = new Command[2];
+      var commandRunCounter = new AtomicInteger(0);
+      // due to the ordering of sets being non-deterministic, we cancel the other command in the
+      // first command and check that the second command is not run
+      Command command1 =
+          new RunCommand(
+              () -> {
+                scheduler.cancel(commands[1]);
+                commandRunCounter.incrementAndGet();
+              });
+      Command command2 =
+          new RunCommand(
+              () -> {
+                scheduler.cancel(commands[0]);
+                commandRunCounter.incrementAndGet();
+              });
+
+      commands[0] = command1;
+      commands[1] = command2;
+
+      scheduler.schedule(command1, command2);
+      scheduler.run();
+
+      assertEquals(
+          1, commandRunCounter.get(), "Second command was run when it shouldn't have been");
+
+      // only one of the commands should be canceled.
+      assertFalse(
+          scheduler.isScheduled(command1) && scheduler.isScheduled(command2),
+          "None of the commands were canceled when one should have been");
+      // one of the commands shouldn't be canceled because the other one is canceled first
+      assertTrue(
+          scheduler.isScheduled(command1) || scheduler.isScheduled(command2),
+          "Both commands were canceled when only one should have been");
+
+      scheduler.run();
+      assertEquals(2, commandRunCounter.get());
+    }
+  }
+
+  @Test
+  void commandKnowsWhenEndedTest() {
+    try (CommandScheduler scheduler = new CommandScheduler()) {
+      Command[] commands = new Command[1];
+      Command command =
+          new FunctionalCommand(
+              () -> {},
+              () -> {},
+              isForced -> {
+                assertFalse(
+                    scheduler.isScheduled(commands[0]),
+                    "Command shouldn't be scheduled when its end is called");
+              },
+              () -> true);
+
+      commands[0] = command;
+      scheduler.schedule(command);
+      scheduler.run();
+      assertFalse(
+          scheduler.isScheduled(command),
+          "Command should be removed from scheduler when its isFinished() returns true");
+    }
+  }
+
+  @Test
+  void scheduleCommandInCommand() {
+    try (CommandScheduler scheduler = new CommandScheduler()) {
+      AtomicInteger counter = new AtomicInteger(0);
+      Command[] commands = new Command[1];
+      Command command1 =
+          new RunCommand(
+              () -> {
+                scheduler.schedule(commands[0]);
+                assertEquals(
+                    1,
+                    counter.get(),
+                    "Command 2's init was not run immediately after getting scheduled");
+              });
+      Command command2 = new InstantCommand(counter::incrementAndGet);
+
+      commands[0] = command2;
```

Comment:
```suggestion
      Command commandToGetScheduled = new InstantCommand(counter::incrementAndGet);
      Command command =
          new RunCommand(
              () -> {
                scheduler.schedule(commandToGetScheduled);
                assertEquals(
                    1,
                    counter.get(),
                    "Scheduled command's init was not run immediately after getting scheduled");
              });
```
Since we don't have any recursive behavior, we can avoid the array. Also changes the naming to match C++.

Reply:
My brain focused too much on the previous test, and I did it this way as well. LMAO.

## Added in 93ad7b8